core/drivers/crypto/caam/hal/imx_6_7/hal_clk.c

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2018-2019 NXP
+ *
+ * Brief   CAAM Clock functions.
+ */
+#include <caam_hal_clk.h>
+#include <caam_io.h>
+#include <mm/core_memprot.h>
+
+/*
+ * Enable/disable the CAAM clocks
+ *
+ * @enable  Enable the clock if true
+ */
+void caam_hal_clk_enable(bool enable)
+{
```

Comment:
This function is a bit messy with all the ifdefs, how about conditionally define three different functions instead?

Reply:
ok